woocommerce/woocommerce
# Unify currency symbol behavior, decode entities in Currency package

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






In most areas where we use currency symbols, we treat them as text; often, we only decode HTML entities with `decodeEntities`, stripping all HTML. However, in `AdvancedFilters > NumberFilter` we allow HTML. 

I think the use case for the component consumer using  HTML there is not worth the inconsistency, confusion, and problems with such symbols ending up in CSV exports, etc.

Technically, it may be a breaking change for `<NumberFilter>`, but I doubt if such behavior was desired in the first place.


### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Checkout this branch
5. `pnpm run build`
5. Run automated tests
7. Customize currency setup
```php
add_action( 'admin_enqueue_scripts', function () {
	wp_add_inline_script( 'wp-hooks', <<<EOD
	wp.hooks.addFilter( 'woocommerce_admin_report_currency', 'dev-blog-example', function() {
		return {
			code: "NZD",
			decimalSeparator: ".",
			precision: 2,
			priceFormat: "%1\$s %2\$s",
			symbol: "&#x62f;.&#x625;",
			symbolPosition: "left",
			thousandSeparator: ",",
		};
	} );
	EOD );
} );
```
8. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomers&filter=advanced`
9. Add filter: `Total spend`
10. Check that the currency is rendered correctly.

![image](https://github.com/user-attachments/assets/1c432d8e-d4e6-48c6-ab63-241d1e1e4a41)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
